src/components/InventoryGroups/Modals/RenameGroupModal.js

```diff
@@ -0,0 +1,86 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import validatorTypes from '@data-driven-forms/react-form-renderer/validator-types';
+import componentTypes from '@data-driven-forms/react-form-renderer/component-types';
+import Modal from './Modal';
+import awesomeDebouncePromise from 'awesome-debounce-promise';
+import { validateGroupName, updateGroupById } from '../utils/api';
+import { nameValidator } from '../helpers/validate';
+import apiWithToast from '../utils/apiWithToast';
+import { useDispatch } from 'react-redux';
+
+const renameGroupSchema = (namePresenceValidator) => ({
+    fields: [
+        {
+            component: componentTypes.TEXT_FIELD,
+            name: 'name',
+            label: 'Group name',
+            helperText:
+          'Can only contain letters, numbers, spaces, hyphens ( - ), and underscores( _ ).',
+            isRequired: true,
+            validate: [
+                namePresenceValidator,
+                { type: validatorTypes.REQUIRED },
+                { type: validatorTypes.MAX_LENGTH, threshold: 50 },
+                nameValidator
+            ]
+        }
+    ]
+});
+
+const RenameGroupModal = ({
+    isModalOpen,
+    setIsModalOpen,
+    modalState = { id: '1', name: 'name' }
+    //remove default values for modalState when you connect it to the table
+}) => {
+    const { id, name } = modalState;
+    const dispatch = useDispatch();
+
+    const handleRenameModal = (values) => {
+        const statusMessages = {
+            onSuccess: {
+                title: 'Success',
+                description: `${name} has been renamed to ${values.name} successfully`
+            },
+            onError: { title: 'Error', description: 'Failed to rename group' }
+        };
+        apiWithToast(dispatch, () => updateGroupById(id, values), statusMessages);
+    };
+
+    const schema = useMemo(() => {
+        const check = async (value) => {
+            const results = await validateGroupName(value);
+            if (results === true) {
+                throw 'Group name already exists';
+            }
+
+            return undefined;
+        };
+
+        // eslint-disable-next-line new-cap
+        const d = awesomeDebouncePromise(check, 500, { onlyResolvesLast: false });
```

Comment:
why we need `onlyResolvesLast` set to false?

Reply:
According to https://www.npmjs.com/package/awesome-debounce-promise this option allows to resolve all promises that were returned and not the "only last returned". This was added in the create group modal - should we remove it?

Reply:
@Fewwy why would you need to resolve all the promises? aren't we interested in only the last one?